### PR TITLE
(SIMP-3018) Add RPM/tarball integration test

### PR DESCRIPTION
**Pull request by jeefberkey**
This commit adds a beaker test that can take either a generated tarball
or the SIMP packagecloud repos and install SIMP, including one agent
from EL6 and EL7.

- Added rpm or tarball integration tests
  - rpm_el7 - EL7 master, EL7 and EL6 agents
  - rpm_el6 - EL6 master, EL7 and EL6 agents

SIMP-3018 #close

### CODE REVIEW

File: spec/acceptance/suites/rpm/01_simp_server_spec.rb [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+  end
+
+  context 'master' do
+    let(:simp_conf_template) { File.read(File.open('spec/acceptance/suites/rpm/files/simp_conf.yaml.erb')) }
+    masters.each do |master|
+      it 'should set up SIMP repositories' do
+        master.install_package('epel-release')
+        # master.install_package('http://yum.puppetlabs.com/puppetlabs-release-pc1-el-7.noarch.rpm')
+        # master.install_package('https://download.postgresql.org/pub/repos/yum/9.4/redhat/rhel-7-x86_64/pgdg-centos94-9.4-3.noarch.rpm')
+        on(master, 'curl -s https://packagecloud.io/install/repositories/simp-project/6_X_Dependencies/script.rpm.sh | bash')
```
Comment (trevor-vaughan) [inline spec/acceptance/suites/rpm/01_simp_server_spec.rb]: We need the ability to point at arbitrary repositories.

File: spec/acceptance/suites/rpm/01_simp_server_spec.rb [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+
+        tarball = find_tarball
+        if tarball.nil? or tarball.empty?
+          warn('='*72)
+          warn('Using Internet repos from packagecloud for testing')
+          warn('Specify a tarball with BEAKER_release_tarball or by placing one in spec/fixtures')
+          warn('='*72)
+          on(host, 'curl -s https://packagecloud.io/install/repositories/simp-project/6_X/script.rpm.sh | bash')
```
Comment (trevor-vaughan) [inline spec/acceptance/suites/rpm/01_simp_server_spec.rb]: Multiple repos here too

File: spec/acceptance/suites/rpm/01_simp_server_spec.rb [hunk 3]
```diff
@@ -0,0 +1,165 @@
...
+        master.install_package('epel-release')
+        # master.install_package('http://yum.puppetlabs.com/puppetlabs-release-pc1-el-7.noarch.rpm')
+        # master.install_package('https://download.postgresql.org/pub/repos/yum/9.4/redhat/rhel-7-x86_64/pgdg-centos94-9.4-3.noarch.rpm')
+        on(master, 'curl -s https://packagecloud.io/install/repositories/simp-project/6_X_Dependencies/script.rpm.sh | bash')
+
+        tarball = find_tarball
+        if tarball.nil? or tarball.empty?
+          warn('='*72)
+          warn('Using Internet repos from packagecloud for testing')
+          warn('Specify a tarball with BEAKER_release_tarball or by placing one in spec/fixtures')
```
Comment (trevor-vaughan) [inline spec/acceptance/suites/rpm/01_simp_server_spec.rb]: If we're going to use the Internet anyway, we can pull tarballs from https://simp-project.com/ISO/SIMP/tar_bundles and we should probably support that method if possible.

File: spec/acceptance/suites/rpm/01_simp_server_spec.rb [hunk 4]
```diff
@@ -0,0 +1,165 @@
...
+  # context 'classify nodes' do
+  # end
+
+  context 'agents' do
+    agents.each do |agent|
+      it 'should install the agent' do
+        agent.install_package('http://yum.puppetlabs.com/puppetlabs-release-pc1-el-7.noarch.rpm')
+        agent.install_package('puppet-agent')
+        agent.install_package('net-tools')
+        on(agent, 'curl -s https://packagecloud.io/install/repositories/simp-project/6_X_Dependencies/script.rpm.sh | bash')
```
Comment (trevor-vaughan) [inline spec/acceptance/suites/rpm/01_simp_server_spec.rb]: Last bit of remote fun.

File: spec/acceptance/suites/rpm/01_simp_server_spec.rb [hunk 5]
```diff
@@ -0,0 +1,165 @@
...
+    end
+  end
+
+  # context 'classify nodes' do
+  # end
+
+  context 'agents' do
+    agents.each do |agent|
+      it 'should install the agent' do
+        agent.install_package('http://yum.puppetlabs.com/puppetlabs-release-pc1-el-7.noarch.rpm')
```
Comment (trevor-vaughan) [inline spec/acceptance/suites/rpm/01_simp_server_spec.rb]: Don't you need to check against EL6 as well?

File: spec/acceptance/suites/rpm/files/simp_conf.yaml.erb [hunk 6]
```diff
@@ -0,0 +1,268 @@
...
+#
+cli::is_ldap_server: true
+
+# === cli::network::dhcp ===
+# Whether to use DHCP to set up your network ("static" or "dhcp").
+cli::network::dhcp: dhcp
+
+# === cli::network::gateway ===
+# The default gateway.
+cli::network::gateway: 10.0.2.2
```
Comment (trevor-vaughan) [inline spec/acceptance/suites/rpm/files/simp_conf.yaml.erb]: Are these addresses always consistent?

File: spec/acceptance/suites/rpm/01_simp_server_spec.rb (hunk 5 above)
Comment (jeefberkey) [inline spec/acceptance/suites/rpm/01_simp_server_spec.rb]: That would need to be another suite, I can't ask machines running this test to run two puppetservers.

File: spec/acceptance/suites/rpm/files/simp_conf.yaml.erb (hunk 6 above)
Comment (jeefberkey) [inline spec/acceptance/suites/rpm/files/simp_conf.yaml.erb]: Yes, in vagrant.

File: spec/acceptance/suites/rpm/01_simp_server_spec.rb (hunk 2 above)
Comment (jeefberkey) [inline spec/acceptance/suites/rpm/01_simp_server_spec.rb]: How would you configure a test like this?

File: spec/acceptance/suites/rpm/01_simp_server_spec.rb (hunk 5 above)
Comment (trevor-vaughan) [inline spec/acceptance/suites/rpm/01_simp_server_spec.rb]: @jeefberkey You can filter based on the client type and install the correct agent. The point is to make sure that all supported clients can be served by the same code from the same server.

We can make this an additional ticket if we need to.

File: spec/acceptance/suites/rpm_el7/01_simp_server_spec.rb [hunk 7]
```diff
@@ -0,0 +1,207 @@
...
+    end
+  end
+
+  # context 'classify nodes' do
+  # end
+
+  context 'agents' do
+    agents.each do |agent|
+      it 'should install the agent' do
+        if agent.host_hash[:platform] =~ /el-7/
```
Comment (trevor-vaughan) [inline spec/acceptance/suites/rpm_el7/01_simp_server_spec.rb]: This needs to be specific in case we update things later, but it's fine for now.

**Pull request merged**